hack/quickstart-setup.sh

```diff
@@ -0,0 +1,394 @@
+  kubectl config use-context kind-${KIND_CLUSTER_CONTROL_PLANE}
+  join=$(clusteradm get token --context kind-${KIND_CLUSTER_CONTROL_PLANE} |  grep -o  'clusteradm.*--cluster-name')
+  kubectl config use-context kind-${clusterName}
+  ${join} kind-${clusterName} --bundle-version='0.11.0' --feature-gates=RawFeedbackJsonString=true --force-internal-endpoint-lookup --context kind-${clusterName} | grep clusteradm
+  echo "accepting OCM spoke cluster invite"
+  kubectl config use-context kind-${KIND_CLUSTER_CONTROL_PLANE}
+  
+  max_retry=18
+  counter=0
+  until clusteradm accept --clusters kind-${clusterName}
+  do
+     sleep 10
+     [[ counter -eq $max_retry ]] && echo "Failed!" && exit 1
+     echo "Trying again. Try #$counter"
+     ((++counter))
+  done
+
+}
+
+initController() {
+    clusterName=${1}
+    kubectl config use-context kind-${clusterName}
+    echo "Initialize local dev setup for the controller on ${clusterName}"
+
+    # Add the mgc CRDs
+    kustomize build ${MGC_REPO}/config/crd | kubectl apply -f -
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: multi-cluster-gateways
+EOF
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ${KIND_CLUSTER_PREFIX}aws-credentials
+  namespace: multi-cluster-gateways
+type: Opaque
+stringData:
+  AWS_ACCESS_KEY_ID: ${AWS_ACCESS_KEY_ID}
+  AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY}
+  AWS_REGION: ${AWS_REGION}
+EOF
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ${KIND_CLUSTER_PREFIX}controller-config
+  namespace: multi-cluster-gateways
+data:
+  AWS_DNS_PUBLIC_ZONE_ID: ${AWS_DNS_PUBLIC_ZONE_ID}
+  ZONE_ROOT_DOMAIN: ${ZONE_ROOT_DOMAIN}
+  LOG_LEVEL: "${LOG_LEVEL}"
+EOF
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: kuadrant.io/v1alpha1
+kind: ManagedZone
+metadata:
+  name: ${KIND_CLUSTER_PREFIX}dev-mz
+  namespace: multi-cluster-gateways
+spec:
+  id: ${AWS_DNS_PUBLIC_ZONE_ID}
+  domainName: ${ZONE_ROOT_DOMAIN}
+  description: "Dev Managed Zone"
+  dnsProviderSecretRef:
+    name: ${KIND_CLUSTER_PREFIX}aws-credentials
+    namespace: multi-cluster-gateways
+    type: AWS
+EOF
+}
+
+
+echo "Verifying that pre-requisite binaries are present on the system."
+
+declare -a BINARY_NAMES=("kustomize" "kind" "helm" "yq" "operator-sdk" "istioctl" "clusteradm")
+declare -a EXPECTED_VERSIONS=("4.5.4" "0.17.0" "3.10.0" "4.30.8" "1.17.0" "1.28.0" "0.6.0")
+
+# Check if a binary is installed and has the expected version
+check_binary_version() {
+    local binary_name="$1"
+    local expected_version="$2"
+    local semver_pattern="$3"
+
+    # Check if the binary is installed
+    if ! command -v "$binary_name" &>/dev/null; then
+        echo "Error: $binary_name not found. Please review the pre-requisite versions in the documentation against your local version of $binary_name"
```

Comment:
Where are those docs?

Reply:
This will be the doc mentioned in the associated issue. The user would have come from there to run the command. We're aiming for something similar to https://open-cluster-management.io/getting-started/quick-start/ so the prereq section should have prepared the user for this.

Reply:
Sounds good

Reply:
I don't think these exist yet. The pre-requisite version numbers should probably be listed in the quickstart guide on the website.

Reply:
I guess we want to include it somewhere under https://docs.kuadrant.io/multicluster-gateway-controller/ ? It's likely a small change so we could do this as part of this issue if folks are happy with how the script is running here.

Reply:
Captured this in the description.